swift/Workflow/Sources/RenderContext.swift

```diff
@@ -158,3 +152,107 @@ extension RenderContext {
     }
 
 }
+
+
+
+
+
+extension Signal: AnyWorkflowConvertible where Error == Never {
+
+    public func asAnyWorkflow() -> AnyWorkflow<Void, Value> {
+        return SignalWorkflow(signal: self).asAnyWorkflow()
+    }
+
+}
+
+extension SignalProducer: AnyWorkflowConvertible where Error == Never {
+
+    public func asAnyWorkflow() -> AnyWorkflow<Void, Value> {
+        return SignalProducerWorkflow(signalProducer: self).asAnyWorkflow()
+    }
+
+}
+
+extension Property: AnyWorkflowConvertible {
+
+    public func asAnyWorkflow() -> AnyWorkflow<Value, Never> {
+        return PropertyWorkflow(property: self).asAnyWorkflow()
+    }
+
+
+}
+
+private struct SignalWorkflow<Value>: Workflow {
+    var signal: Signal<Value, Never>
+    typealias Output = Value
+    typealias State = Void
+    func makeInitialState() -> State {
+        return ()
+    }
+    func workflowDidChange(from previousWorkflow: SignalWorkflow, state: inout State) {
+    }
+    typealias Rendering = Void
+    func render(state: State, context: RenderContext<SignalWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: AnyWorkflowAction.self)
+        context.runSideEffect(key: UUID()) { [signal] lifetime in
+            signal
+                .take(during: lifetime)
+                .map { AnyWorkflowAction(sendingOutput: $0) }
+                .observe(on: QueueScheduler.workflowExecution)
+                .observeValues(sink.send)
+        }
+        return ()
+    }
+}
+
+private struct SignalProducerWorkflow<Value>: Workflow {
+    var signalProducer: SignalProducer<Value, Never>
+    typealias Output = Value
+    typealias State = Void
+    func makeInitialState() -> State {
+        return ()
+    }
+    func workflowDidChange(from previousWorkflow: SignalProducerWorkflow, state: inout State) {
+    }
+    typealias Rendering = Void
+    func render(state: State, context: RenderContext<SignalProducerWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: AnyWorkflowAction.self)
+        context.runSideEffect(key: UUID()) { [signalProducer] lifetime in
```

Comment:
Isn't this going to restart the `SignalProducer` on every render pass?

Reply:
Ah, yep! I copied too much from SignalWorkflow. Good catch.